Raspi/DAS.js

```diff
@@ -0,0 +1,175 @@
+};
+const mqttClient = mqtt.connect('mqtt://localhost:1883', mqttOptions);
+
+/* Start of main code */
+
+function connectWithTeensy() {
+  // Open event to tell us when connection with teensy has been made
+  serialPort.open((err) => {
+    if (err) {
+      winston.error(err.message);
+    } else {
+      winston.info('Serial port open');
+    }
+  });
+
+  serialPort.write('reading', (err) => {
+    if (err) {
+      winston.error('Error on write: ', err.message);
+    } else {
+      winston.info('Reading Teensy Data!');
+    }
+  })
+}
+
+// Check if ant-plus dongle is connected to Raspberry Pi
+function checkAntPlusConnection() {
+  if (!args['antplus'] && !antPlus.open()) {
+    winston.error('Ant-plus usb stick not found!');
+    return false;
+  }
+  // Stop looking for ant-plus dongle
+  clearInterval(findAntPlusInterval);
+  setImmediate(connectWithTeensy);
+  return true;
+}
+
+// Only ran if connected to mqtt broker
+function mqttConnected() {
+  findAntPlusInterval = setInterval(() => {
+    winston.info('Finding ant-plus USB...');
+    checkAntPlusConnection();
+  }, 1000);
+}
+
+// Receive filename for us to use
+let IS_RECORDING = false;
+let INITIAL_TIME = 0;
+function main(topic, message) {
+  winston.info(`Topic fired: ${topic}`);
+  switch (topic) {
+    // Filename to use
+    case 'filename':
+      FILENAME = message.toString();
+      IS_RECORDING = true;
+      break;
+    default:
+      winston.error(`Unexpected topic: ${topic}`);
+  }
+}
+
+winston.info('Connecting to mqtt server...');
+mqttClient.subscribe('filename');
+mqttClient.on('connect', mqttConnected);
+mqttClient.on('message', main);
+
+antPlus.on('startup', () => {
+  winston.info('ant-plus stick initialized');
+  // Connect to the first device found
+  bicyclePowerSensor.attach(0, 0);
+});
+
+let CADENCE = 0;
+let POWER = 0;
+serialPort.on('open', () => {
+  winston.info('Port opened with Teensy');
+  // Power meter data
+  bicyclePowerSensor.on('powerData', (data) => {
+    if (IS_RECORDING) {
+      // Store power meter into global variable
+      CADENCE = data.Cadence;
+      POWER = data.Power;
+      winston.info(`ID: ${data.DeviceID}, Cadence: ${CADENCE}, Power: ${POWER}`);
+    }
+  });
+
+  // mqtt broker and Teensy need to be both connected to receive data
+  parser.on('data', (data) => {
+    winston.info(`Teensy - ${data}`);
+    if (data === 'start') {
+      // Let mqtt broker know we have started recording
+      mqttClient.publish('start');
+      INITIAL_TIME = Math.floor(Date.now());
+    } else if (data === 'stop') {
+      // Reset variables
+      IS_RECORDING = false;
+      CADENCE = 0;
+      POWER = 0;
+      mqttClient.publish('stop');
+    }
+
+    if (IS_RECORDING) {
+      const currentTime = Math.floor(Date.now());
+      const outputData = `${data}&filename=${FILENAME}&time=${(currentTime - INITIAL_TIME)}&power=${POWER}&cadence=${CADENCE}`;
+      mqttClient.publish('data', outputData);
+    }
+  });
+});
```

Comment:
1. Definitely need linters.
2. Should break down file into multiple modules

Reply:
Definitely the next steps to do. But will need to update master branch with what we have at the moment. Otherwise the PR's get too big (like this one).